src/sync/impl/sync_client.hpp

```diff
@@ -42,6 +46,24 @@ struct SyncClient {
         client.run();
     }) // Throws
     {
+#if REALM_PLATFORM_APPLE
```

Comment:
Presumably Android has an equivalent way of observing network reachability. Shouldn't we use it in the same way?